### PR TITLE
feat: add newtonsoft serialize functionality

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
I had a use-case where I needed to use a different library for serializing the models to JSON. This PR adds support for newtonsoft through a new parameter.

### CODE REVIEW

File: components/channel/publish.js [hunk 1]
```diff
@@ -27,13 +27,14 @@ function getPublishCode(publishMessage, realizedChannelPath) {
  * @param {*} channelName 
  * @param {*} channelParameters 
  * @param {*} publishMessage 
+ * @param {*} params 
  */
-export default function publish(channelName, channelParameters, publishMessage) {
+export default function publish(channelName, channelParameters, publishMessage, params) {
```
Comment (magicmatatjahu) [inline components/channel/publish.js:32]: I don't see any place where you use `params` argument in this function.

File: components/channel/publish.js [hunk 2]
```diff
@@ -27,13 +27,14 @@ function getPublishCode(publishMessage, realizedChannelPath) {
...
+ * @param {*} params 
  */
-export default function publish(channelName, channelParameters, publishMessage) {
+export default function publish(channelName, channelParameters, publishMessage, params) {
   const functionParameters = getFunctionParameters(publishMessage, channelParameters);
   const realizedChannelPath = realizeChannelName(channelParameters, channelName);
   const publishCode = getPublishCode(publishMessage, realizedChannelPath);
   return `
-${serializer(publishMessage)}
+${serializer(publishMessage, channelParameters)}
```
Comment (magicmatatjahu) [inline components/channel/publish.js:37]: I guess, it should be this:

```suggestion
${serializer(publishMessage, params)}
```

File: components/channel/subscribe.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ export default function subscribe(channelName, channelParameters, subscriptionMe
   const subscribeParameters = getSubscribeParameters(realizedChannelPath, queue);
 
   return `
-  ${deserializer(subscriptionMessage)}
+  ${deserializer(subscriptionMessage, channelParameters)}
```
Comment (magicmatatjahu) [inline components/channel/subscribe.js:45]:
```suggestion
  ${deserializer(subscriptionMessage, params)}
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
Probably I found bugs.

File: components/channel/publish.js (hunk 1 above)
Comment (jonaslagoni) [reply components/channel/publish.js:32]: Yep, removed 😄

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**